src/dogear.js

```diff
@@ -5,141 +5,133 @@ const Oppsy = require('oppsy');
+  'proc.delay': {
+    path: 'psdelay',
+    type: 'timing'
   }
+};
 
-  this.client = new StatsD(this._settings.statsdConfig);
-}
-
-
-internals.Dogear.prototype.close = close;
-function close(cb) {
-  if (this._ops) {
-    this._ops.stop();
-  }
-  this.client.close(cb);
-}
+class Dogear {
+  constructor(config) {
+    this._config = config;
 
+    this._settings = Hoek.applyToDefaults(defaultConfig, config || {});
 
-internals.Dogear.prototype.attachToServer = attachToServer;
-function attachToServer(server) {
+    if (!this._settings.opsMetrics) {
+      this._settings.opsMetrics = Object.keys(opsMetricTypes);
+    }
 
-  const self = this;
+    this.client = new StatsD(this._settings.statsdConfig);
+  }
 
-  server.ext('onPreResponse', (request, reply) => {
-    const msec = Date.now() - request.info.received;
+  close(cb) {
+    if (this._ops) {
+      this._ops.stop();
+    }
+    this.client.close(cb);
+  }
 
-    let path = request._route.path;
-    const specials = request.connection._router.specials;
+  attachToServer(server) {
+    server.ext('onPreResponse', (request, h) => {
+      const msec = Date.now() - request.info.received;
 
-    if (request._route === specials.notFound.route) {
-      path = '/{notFound*}';
-    } else if (specials.options && request._route === specials.options.route) {
-      path = '/{cors*}';
-    } else if (request._route.path === '/' && request._route.method === 'options'){
-      path = '/{cors*}';
-    }
+      let path = request._route.path;
+      const specials = request._core.router.specials;
 
-    const statusCode = (request.response.isBoom) ?
-      request.response.output.statusCode :
-      request.response.statusCode;
+      if (request._route === specials.notFound.route) {
+        path = '/{notFound*}';
+      } else if (specials.options && request._route === specials.options.route) {
+        path = '/{cors*}';
+      } else if (request._route.path === '/' && request._route.method === 'options') {
+        path = '/{cors*}';
+      }
 
-    const tags = [
-      `path:${ path }`,
-      `method:${ request.method.toUpperCase() }`
-    ];
+      const statusCode = request.response.isBoom ?
+        request.response.output.statusCode :
+        request.response.statusCode;
 
-    if (statusCode) {
-      tags.push(`status:${ statusCode }`);
+      const tags = [
+        `path:${ path }`,
+        `method:${ request.method.toUpperCase() }`
+      ];
 
-      self.client.increment(`request.status.${ statusCode }`, 1, tags);
-      self.client.increment('request.received', 1, tags);
-    }
+      if (statusCode) {
+        tags.push(`status:${ statusCode }`);
 
-    self.client.histogram('request.response_time', msec, tags);
+        this.client.increment(`request.status.${ statusCode }`, 1, tags);
+        this.client.increment('request.received', 1, tags);
+      }
 
-    reply.continue();
-  });
+      this.client.histogram('request.response_time', msec, tags);
 
+      return h.continue;
```

Comment:
This appears to be the only functional change in this entire file. Please revert all other changes and leave this line as `return reply.continue;`